Test Shipping related models and schema

Jeshan7 opened the pull request:
## Why?
  In Shipping_category (Model)
   - Test for get_with_rules/1 was not present.

  In Shipping_method (Model)
   - Tests for get/1, get_all/0, delete/1 and for_package_query/2
     were not present.

  In ShippingRuleIdentifier (Schema)
   - The test file was not present.

  Shipping identifier schema file name was wrong.

## This change addresses the need by:
  In Shipping_category (Model)
   - Added test for get_with_rules/1.

  In Shipping_method (Model)
   - Added tests for get/1, get_all/0, delete/1 and for_package_query/2 .

  In ShippingRuleIdentifier (Schema)
   - Created a new test file.

  Renamed the file schema/shipping_rules/shipping_rule_idenitifier.ex
  to schema/shipping_rules/shipping_rule_identifier.ex


[delivers #163586461]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

Jeshan7 committed:
[#163586461]

Prior to this change,
  In Shipping_category (Model)
   - Test for get_with_rules/1 was not present.

  In Shipping_method (Model)
   - Tests for get/1, get_all/0, delete/1 and for_package_query/2
     were not present.

  In ShippingRuleIdentifier (Schema)
   - Test file was not present.

  Shipping identifier schema file name was wrong.
This change
  In Shipping_category (Model)
   - Added test for get_with_rules/1.

  In Shipping_method (Model)
   - Added tests for get/1, get_all/0, delete/1 and for_package_query/2 .

  In ShippingRuleIdentifier (Schema)
   - Created a new test file.

  Renamed the file schema/shipping_rules/shipping_rule_idenitifier.ex
  to schema/shipping_rules/shipping_rule_identifier.ex

arjun289 commented on apps/snitch_core/test/data/model/shipping_category_test.exs:
```diff
@@ -36,6 +36,19 @@ defmodule Snitch.Data.Model.ShippingCategoryTest do
+      shipping_rule = insert(:shipping_rule)
+      id = shipping_rule.shipping_category_id
+      {:ok, sc} = ShippingCategory.get_with_rules(id)
+      assert sc.id == id
```
Also, check if rules are returned.

arjun289 commented on apps/snitch_core/test/data/model/shipping_method_test.exs:
```diff
@@ -65,4 +70,82 @@ defmodule Snitch.Data.Model.ShippingMethodTest do
+         country_zone_count: 1,
+         shipping_category_count: 1
+
+    setup :shipping_methods
```
shouldn't setup be above tag for the test

arjun289 commented on apps/snitch_core/test/data/schema/shipping_rules/shipping_rule_identifier_test.exs:
```diff
@@ -0,0 +1,58 @@
+  end
+
+  describe "changeset/2" do
+    test "successful for valid params", %{params: params} do
```
keep the names for changeset function related statements as returns valid or invalid always.

Jeshan7 committed:
[#163586461]

Prior to this change,
  In shipping_category_test (Model)
   - checks for rules were not present
  In shipping_method_test (Model)
   - One of the setup was present below the tag
  Need to change a test statement
  Test for shipping rule were not present
This change
  In shipping_category_test (Model)
   - checks for rules is now  present.
  In shipping_method_test (Model)
   - deleted that setup as alreay one is present at top.
  changed a test statement(shipping rule identifier)

  Created a new file for shipping rule test.

   get/1 function in shipping_rule.ex (model) was taking wrong arguments.
   Updated this function

joaothallis commented on apps/snitch_core/lib/core/data/model/shipping_rule.ex:
```diff
@@ -25,8 +25,14 @@ defmodule Snitch.Data.Model.ShippingRule do
+        {:ok, shipping_rule}
+
+      {:error, msg} ->
+        {:error, msg}
```
Using with do not need of lines 33 and 34

Jeshan7 replied:
@joaothallis Thank you so much for your invaluable comment. Now, get/1 uses with statement  which handles the error clause .

arjun289 commented on apps/snitch_core/test/data/model/shipping_method_test.exs:
```diff
@@ -65,4 +70,79 @@ defmodule Snitch.Data.Model.ShippingMethodTest do
+         country_zone_count: 1,
+         shipping_category_count: 1
+    test "successfully deletes a shipping method", %{shipping_methods: [sm]} do
+      sm_deleted = ShippingMethod.delete(sm.id)
```
This will generate unused `sm_deleted` warning. Also, pattern match on the result returned I think it is `{:ok, _sm}`

arjun289 commented on apps/snitch_core/test/data/model/shipping_rule_test.exs:
```diff
@@ -0,0 +1,43 @@
+  describe "get_all_by_shipping_category/1" do
+    test "returns a shipping_rule", %{shipping_rule: shipping_rule} do
+      id = shipping_rule.shipping_category_id
+      [new_shipping_rule] = ShippingRule.get_all_by_shipping_category(id)
```
`new_shipping_rule` can be modified to something like the result or returned_shipping_rule new feels as if something is being created.

arjun289 reviewed (commented):
Also, can you try to explore how to test `identifier_with_module` in shipping_rule_identifier schema?

Jeshan7 committed:
[#163586461]
Prior to this change,
 In shipping_method_test.exs
   - sm_deleted was not needed on line 114.
  In shipping_rule_test.exs
   - new_shipping_rule needs to be converted.
  In model/shipping_rule.ex, model/shipping_category.ex
   - get/1 needed optimisation

This change
  In shipping_method_test.exs
   - Line 114 is converted to {:ok, _}.
  In shipping_rule_test.exs
   - new_shipping_rule word is converted to returned_shipping_rule.
  In model/shipping_rule.ex , model/shipping_category.ex
   - get/1 now uses with statement.

Jeshan7 committed:
[#163586461]

Prior to this change,
  - No test case was present for identifier_with_module/0 in
     shipping_rule_identifier_test.exs

This change
  - Test case is now present for identifier_with_module/0 in
    shipping_rule_identifier_test.exs

arjun289 committed:
Merge branch 'develop' into shipping_test
